### DISCUSSION TITLE
Adding baremetal / dedicated nodes

### DISCUSSION THREAD

**Discussion by DaanVanVugt** [vitobotta/hetzner-k3s#231] [Q&A]
I am trying to add one of our dedicated nodes, and managed to register it, following guides:

- https://pet2cattle.com/2021/04/k3s-join-nodes
- https://docs.hetzner.com/cloud/networks/connect-dedi-vswitch/

The node is listed in `kubectl get nodes`, but has a taint: `node.kubernetes.io/network-unavailable:NoSchedule`

This seems to be caused by

```
Warning  FailedToCreateRoute      4m42s (x123 over 65m)
 route_controller  (combined from similar events):
   Could not create route 377627a0-3d0b-4aa0-977a-b475189eb2c2 10.244.1.0/24
  for node benchy after 164.80536ms: hcloud/CreateRoute: hcops/AllServersCache.ByName: benchy hcops/AllServersCache.getCache: not found
```

which makes sense, since it is not a VPS and so the hcloud APIs will not like it.

Is there a way to prevent this taint from occurring? Or to manually make this route?

**Comment by vitobotta** [comment]
Can you see the dedicated node in the private network from one of the regular nodes? Converting this into a discussion since it's not an issue with the tool.

**Reply by DaanVanVugt** [reply]
I can ping back and forth between the VPS master and the dedicated node, to the IP registered with the vswitch (10.0.1.2 dedi, 10.0.0.2 vps). There is also a `flannel.1` adapter which has an ip `10.244.1.0` which I cannot ping from the VPS.

**Reply by vitobotta** [reply]
Not sure what could be the problem, unfortunately I haven't experimented with this kind of setup. Hopefully someone can help :)

**Reply by DaanVanVugt** [reply]
I think it is related to https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/433 and https://github.com/hetznercloud/hcloud-cloud-controller-manager/issues/330

**Reply by DaanVanVugt** [reply]
It seems there are several forks of the hcloud-cloud-controller-manager which promise to support root / baremetal / dedicated nodes. Would it make sense to make the version of the cloud-controller-manager configurable?

**Reply by vitobotta** [reply]
I have on my list to make the versions configurable but this would require to make it possible to change the source as well.

**Reply by DaanVanVugt** [reply]
Indeed. I have made a PR at https://github.com/vitobotta/hetzner-k3s/pull/233

**Reply by vitobotta** [reply]
@DaanVanVugt Thanks!

**Reply by Aubermean** [reply]
@DaanVanVugt  @vitobotta the latest version of CCM released in December supports Robot/dedicated servers.
